Userland/Utilities/tar.cpp

```diff
@@ -104,39 +97,29 @@ int main(int argc, char** argv)
                 case Archive::TarFileType::AlternateNormalFile: {
                     Core::File::ensure_parent_directories(absolute_path);
 
-                    int fd = open(absolute_path.characters(), O_CREAT | O_WRONLY, header.mode());
-                    if (fd < 0) {
-                        perror("open");
-                        return 1;
-                    }
+                    int fd = TRY(Core::System::open(absolute_path.characters(), O_CREAT | O_WRONLY, header.mode()));
 
                     Array<u8, buffer_size> buffer;
-                    size_t nread;
-                    while ((nread = file_stream.read(buffer)) > 0) {
-                        if (write(fd, buffer.data(), nread) < 0) {
-                            perror("write");
-                            return 1;
-                        }
-                    }
+                    while (file_stream.read(buffer) > 0)
+                        TRY(Core::System::write(fd, buffer.span()));
+
                     close(fd);
                     break;
                 }
                 case Archive::TarFileType::SymLink: {
                     Core::File::ensure_parent_directories(absolute_path);
 
-                    if (symlink(header.link_name().to_string().characters(), absolute_path.characters())) {
-                        perror("symlink");
-                        return 1;
-                    }
-
+                    TRY(Core::System::symlink(header.link_name().to_string().characters(), absolute_path.characters()));
```

Comment:
Why should a Core:System API require char*?

Reply:
Fixed.